src/edu/csus/ecs/pc2/core/model/Problem.java

```diff
@@ -1992,31 +2008,11 @@ public void setVivaInputValidatorHasBeenRun(boolean vivaInputValidatorHasBeenRun
         vivaSettings.setVivaInputValidatorHasBeenRun(vivaInputValidatorHasBeenRun);
     }
 
-    /**
-     * Returns the flag indicating whether this Problem has a runtime memory limit associated with it.
-     * 
-     * @return the flag indicating whether or not this Problem has been configured with a memory limit.
-     */
-    public boolean hasMemoryLimit() {
-        return hasMemoryLimit;
```

Comment:
Can we remove the field hasMemoryLimit and change this method just...
return memoryLimitMB  > 0;
?

Reply:
Sure, we COULD do that.  I tend to want to isolate functions with accessors.  What if we later decide we have a different definition of what it means to "have a memory limit"?   If we keep it surrounded by an accessor -- and all other code always uses the accessor (which I think I tried to ensure), we only have to change the accessor to make such a change.  I think it's worth the one-line accessor to provide this improved code structure.

Reply:
Since I did not suggest removing or changing the sig,   My suggestion was to change the code inside
the accessor which is supported by your comment.   Per the comment for Problem:40
 public static final int DEFAULT_MEMORY_LIMIT_MB = 0 ;   //zero memory limit = "none", i.e. the problem can use all available memory
that indicates to me that memoryLimitMB > 0; is the condition for hasMemoryLimit.

Reply:
Ok, I misinterpreted your comment.  But:  there is something wrong.  The entire method `hasMemoryLimit()` was removed entirely.  You are commenting about code that was removed.